stress/client/main.go

```diff
@@ -0,0 +1,323 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/interop"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+	metricspb "google.golang.org/grpc/stress/grpc_testing"
+)
+
+var (
+	serverAddressesPtr      = flag.String("server_addresses", "localhost:8080", "a list of server addresses")
+	testCasesPtr            = flag.String("test_cases", "", "a list of test cases along with the relative weights")
+	testDurationSecsPtr     = flag.Int("test_duration_secs", -1, "test duration in seconds")
+	numChannelsPerServerPtr = flag.Int("num_channels_per_server", 1, "Number of channels (i.e connections) to each server")
+	numStubsPerChannelPtr   = flag.Int("num_stubs_per_channel", 1, "Number of client stubs per each connection to server")
+	metricsPortPtr          = flag.Int("metrics_port", 8081, "The port at which the stress client exposes QPS metrics")
+)
+
+// testCaseType is the type of test to be run
+type testCaseType uint32
+
+const (
+	// emptyUnary is to make a unary RPC with empty request and response
+	emptyUnary testCaseType = 0
+
+	// largeUnary is to make a unary RPC with large payload in the request and response
+	largeUnary testCaseType = 1
+
+	// TODO largeCompressedUnary
+
+	// clientStreaming is to make a client streaming RPC
+	clientStreaming testCaseType = 3
+
+	// serverStreaming is to make a server streaming RPC
+	serverStreaming testCaseType = 4
+
+	// emptyStream is to make a bi-directional streaming with zero message
+	emptyStream testCaseType = 5
+
+	// unknownTest means something is wrong
+	unknownTest testCaseType = 6
+)
+
+var testCaseNameTypeMap = map[string]testCaseType{
+	"empty_unary":      emptyUnary,
+	"large_unary":      largeUnary,
+	"client_streaming": clientStreaming,
+	"server_streaming": serverStreaming,
+	"empty_stream":     emptyStream,
+}
+
+var testCaseTypeNameMap = map[testCaseType]string{
+	emptyUnary:      "empty_unary",
+	largeUnary:      "large_unary",
+	clientStreaming: "client_streaming",
+	serverStreaming: "server_streaming",
+	emptyStream:     "empty_stream",
+}
+
```

Comment:
All these are awkward. I think you can define 

type testCase struct {
  name string
  weight int
}

Then build a slice of testCase from testCaseString. You do not need to have map for fast accessing. The slice size is very small